### ISSUE TITLE
Fontify undeclared formal parameters.

### ISSUE THREAD

**Issue by bradyt** [thedailyshow/dart-mode#18]
While fontifying implicitly declared variables in general seems difficult (wrong? impossible?), it should be easy to fontify undeclared formal parameters. It is even idiomatic Dart in some places, for example, from [AVOID annotating inferred parameter types on function expressions.](https://www.dartlang.org/guides/language/effective-dart/design#avoid-annotating-inferred-parameter-types-on-function-expressions)

> ``` dart
> var names = people.map((person) => person.name);
> ```

Then we would fontify the first appearance of `person`, for the lexical scope of the formal parameters.

In other words, we would extend the emacs fontification idiom, from only declared variables and parameters, to all variables and parameters which obviously have a first appearance in their scope. That is, we would include undeclared formal parameters.
